### PR TITLE
Remove warnings / Change some design stuff

### PR DESCRIPTION
**Pull request by Miniplop**
- Remove the warnings and configErrors from the serverState (not used anymore by the frontend)
- Remove the warnings page
- Remove the log in warning
- Refactor the env store
- Add loader on the slice and ct empty state

### CODE REVIEW

File: packages/slice-machine/pages/_app.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ function MyApp({ Component, pageProps }: AppContext & AppInitialProps) {
     }
 
     const { store, persistor } = configureStore({
-      environment: { env: serverState.env, warnings: [], configErrors: {} },
+      environment: serverState.env,
```
Comment (Heyrows) [inline packages/slice-machine/pages/_app.tsx:65]: nice

File: packages/slice-machine/server/src/api/state.ts [hunk 2]
```diff
@@ -104,7 +76,11 @@ export default async function handler(
   };
 
   return {
-    ...serverState,
+    customTypes: serverState.customTypes,
+    remoteCustomTypes: serverState.remoteCustomTypes,
+    libraries: serverState.libraries,
+    remoteSlices: serverState.remoteSlices,
+    clientError: serverState.clientError,
```
Comment (Heyrows) [inline packages/slice-machine/server/src/api/state.ts]: do we still need to send this property ?
It seems only used in the backend on the push handler.

File: packages/slice-machine/src/modules/environment/index.ts [hunk 3]
```diff
@@ -80,15 +69,15 @@ export const getIsTrackingAvailable = (
...
+    store.environment.manifest.tracking === undefined ||
+    store.environment.manifest.tracking
   );
 };
 
 export const getStorybookUrl = (
   store: SliceMachineStoreType
 ): string | null => {
-  return store.environment.env.manifest.storybook || null;
+  return store.environment.manifest.storybook || null;
```
Comment (Heyrows) [inline packages/slice-machine/src/modules/environment/index.ts:80]: seems way better

File: packages/slice-machine/src/modules/environment/index.ts [hunk 4]
```diff
@@ -141,9 +130,7 @@ export const environmentReducer: Reducer<
     case getType(refreshStateCreator):
       return {
         ...state,
-        env: action.payload.env,
-        warnings: action.payload.warnings,
-        configErrors: action.payload.configErrors,
+        ...action.payload.env,
```
Comment (Heyrows) [inline packages/slice-machine/src/modules/environment/index.ts:133]: are you sure we still send an object with an env property and not the end directly in the payload ?

File: packages/slice-machine/src/modules/environment/types.ts [hunk 5]
```diff
@@ -1,9 +1,3 @@
 import { FrontEndEnvironment } from "@lib/models/common/Environment";
-import Warning from "@models/common/Warning";
-import { ConfigErrors } from "@models/server/ServerState";
 
-export type EnvironmentStoreType = {
-  env: FrontEndEnvironment;
-  warnings: ReadonlyArray<Warning>;
-  configErrors: ConfigErrors;
-};
+export type EnvironmentStoreType = FrontEndEnvironment;
```
Comment (Heyrows) [inline packages/slice-machine/src/modules/environment/types.ts:3]: Was it for the naming coherence or to not rename everythign that you decided to keep that ?

**Review by Heyrows — CHANGES_REQUESTED** [review]

File: packages/slice-machine/pages/_app.tsx (hunk 1 above)
Comment (Miniplop) [reply packages/slice-machine/pages/_app.tsx:65]: cool :)

File: packages/slice-machine/server/src/api/state.ts (hunk 2 above)
Comment (Miniplop) [reply packages/slice-machine/server/src/api/state.ts]: I think we can can also remove it

File: packages/slice-machine/src/modules/environment/index.ts (hunk 3 above)
Comment (Miniplop) [reply packages/slice-machine/src/modules/environment/index.ts:80]: coool 👍

File: packages/slice-machine/src/modules/environment/types.ts (hunk 5 above)
Comment (Miniplop) [reply packages/slice-machine/src/modules/environment/types.ts:3]: Just for keeping the name convention for store type

File: packages/slice-machine/src/modules/environment/index.ts (hunk 4 above)
Comment (Miniplop) [reply packages/slice-machine/src/modules/environment/index.ts:133]: Yes I'm sure look a the the refresh state action :),

**Review by Heyrows — APPROVED** [review]

**Pull request merged**